### DISCUSSION TITLE
Verification of Secure Element generated dilithium2 signature with openssl/oqs provider library

### DISCUSSION THREAD

**Discussion by Vodi75** [open-quantum-safe/liboqs#1776] [Q&A]
Dear All, 
Please apologize all my mistakes in this new discussion as a newbie here at github community if any. 
I was playing around with openssl 3.3 + oqs provider library supportind pqc algorithms and all works fine however I got an interesting problem: on a Java SE installed an application (got from a 3rd party) which can generate dilithium 2/3/4/5 keypairs and drop a signature based on an input got from me (hex coded string in an APDU) and perform a verification on the result as well. My question is how can I verify the result in openssl if the java applet does not use any hash algoritzm definition in its code. No input to define it and also in the java code no sign of hardcoded hash algo, however the built in veif of the SE applet shows the correctness of the signature. If I properly format the plain keys into pem formatted data openssl can use the keypair for signature and verify action so I think the keys are ok. Currently I think the issue is coming from the hash algo difference... Any suggestion maybe, please? Thanks in advance!

**Comment by baentsch** [comment]
By default oqs-provider AFAIK uses no hash algo for plain PQ algs. Please specify exactly which openssl command(s) you tried, what worked and what not (stating what you expect).
